process/block/shardblock.go

```diff
@@ -1258,7 +1258,7 @@ func (sp *shardProcessor) snapShotEpochStartFromMeta(header data.ShardHeaderHand
 				log.Debug("using scheduled root hash for snapshotting", "schRootHash", schRootHash)
 				rootHash = schRootHash
 			}
-			epoch := header.GetEpoch()
+			epoch := sp.epochStartTrigger.MetaEpoch()
```

Comment:
I think this is pretty solid as the rollback of the epoch start metablock is not possible in this case. I remember that the epoch start trigger triggers the new epoch after it has the confirmation metablock